great_expectations/experimental/datasources/data_asset/data_connector/s3_data_connector.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, List, Optional
+
+from great_expectations.core.batch_spec import PathBatchSpec, S3BatchSpec
+from great_expectations.datasource.data_connector.util import (
+    list_s3_keys,
+    sanitize_prefix_for_s3,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.file_path_data_connector import (
+    FilePathDataConnector,
+)
+
+try:
+    import boto3
+except ImportError:
+    boto3 = None
+
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+
+
+logger = logging.getLogger(__name__)
+
+
+class S3DataConnector(FilePathDataConnector):
+    """Extension of FilePathDataConnector used to connect to S3.
+
+
+    Args:
+        name (str): required name for DataConnector
+        datasource_name (str): required name for datasource
+        bucket (str): bucket for S3
+        regex (dict): regex configuration for filtering data_references
+        prefix (str): S3 prefix
+        delimiter (str): S3 delimiter
+        max_keys (int): S3 max_keys (default is 1000)
+        boto3_options (dict): optional boto3 options
+        # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+        # TODO: <Alex>ALEX</Alex>
+        # sorters (list): optional list of sorters for sorting data_references
+        # TODO: <Alex>ALEX</Alex>
+    """
+
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
```

Comment:
I actually don't see datasource_name or data_asset_name used in any of these implementations; we could remove them

Reply:
@donaldheppner Please see https://github.com/great-expectations/great_expectations/pull/7094#discussion_r1103664521 for a detailed explanation.  We can remove the `name` (I raised a question above regarding whether or not we want to do it -- I thought that it could help make troubleshooting easier), but we definitely cannot remove `datasource_name` (essential for `Batch` query logic).  Thanks.